tests/cached_path_test.py

```diff
@@ -327,12 +328,24 @@ def test_cached_path_extract_remote_zip(self):
 
 class TestCachedPathGs(BaseTestClass):
     @flaky
-    def test_cache_blob(self):
+    def test_bytes_and_cache_blob(self):
+        # Get some bytes without downloading the file.
+        bytes_snippet = get_bytes_range(
+            "gs://allennlp-public-models/bert-xsmall-dummy.tar.gz", 5, 10
+        )
+        assert len(bytes_snippet) == 10
+
         path = cached_path("gs://allennlp-public-models/bert-xsmall-dummy.tar.gz")
         assert path.is_file()
         meta = Meta.from_path(_meta_file_path(path))
         assert meta.etag is not None
 
+        # Now get a range of bytes, this time it should read from the cached file.
+        bytes_snippet_2 = get_bytes_range(
+            "gs://allennlp-public-models/bert-xsmall-dummy.tar.gz", 5, 10
+        )
+        assert bytes_snippet_2 == bytes_snippet
+
```

Comment:
You could add a test for HTTPClient that verifies that get_bytes_range throws a NotImplementedError

Reply:
`get_bytes_range()` won't throw that error, only the client, which causes `get_bytes_range()` to fall back to downloading the whole file.